src/app/Http/Controllers/UserController.php

```diff
@@ -23,7 +23,16 @@ public function show(): View
 
     public function home(): View
     {
-        $posts = Post::latest()->paginate(4);
+        $user = User::find(auth()->id());
+
+        if (!$user) {
+            return view('user.home', ['posts' => collect()]);
+        }
+
+        $followingIds = $user->followings->pluck('id')->push($user->id);
+
+        $posts = Post::whereIn('user_id', $followingIds)->latest()->paginate(4);
```

Comment:
too many queries. you can optimized this more.

Reply:
I have updated my PR, sir. Thank you 🙇